### PR TITLE
Assignment 6

### PR DESCRIPTION
**Pull request by pnotequalnp**

### CODE REVIEW

**Comment by StevenZuelke** [comment]
I Will review this

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 1]
```diff
@@ -28,12 +30,11 @@ public static void ConfigureServices(IServiceCollection services)
 
             services.AddDbContext<ApplicationDbContext>(options =>
                 options.EnableSensitiveDataLogging()
-                       .UseSqlite("Data Source=SecretSanta.db"));
+                       .UseSqlite(Configuration.GetConnectionString("DefaultConnection")));
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Used value correctly

File: SecretSanta/src/SecretSanta.Api/Program.cs [hunk 2]
```diff
@@ -16,7 +17,7 @@ public static void Main(string[] args)
             {
                 IServiceProvider services = scope.ServiceProvider;
                 using var context = services.GetRequiredService<ApplicationDbContext>();
-                context.Database.EnsureCreated();
+                context.Database.Migrate();
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Api/Program.cs]: Changed to Migrate

File: SecretSanta/src/SecretSanta.Data/Migrations/20200216235452_InitialDb.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace SecretSanta.Data.Migrations
+{
+    public partial class InitialDb : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "Groups",
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Data/Migrations/20200216235452_InitialDb.cs]: Added the generation of migrations

File: SecretSanta/src/SecretSanta.Web/Startup.cs [hunk 4]
```diff
@@ -28,9 +34,13 @@ public static void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
-            app.UseStaticFiles();
-
-            app.UseMvcWithDefaultRoute();
+            app.UseStaticFiles();
+
+            app.UseRouting();
+
+            app.UseEndpoints(endpoint =>
+                endpoint.MapDefaultControllerRoute()
+            );
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/Startup.cs]: Updated to 3.x routing

File: SecretSanta/src/SecretSanta.Web/Startup.cs [hunk 5]
```diff
@@ -9,13 +10,18 @@ namespace SecretSanta.Web
...
+
+        public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc(options => options.EnableEndpointRouting = false);
+            services.AddControllersWithViews();
 
-            services.AddHttpClient("SecretSantaApi");
+            services.AddHttpClient("SecretSantaApi", options =>
+                options.BaseAddress = new Uri(Configuration["ApiUrl"])
+            );
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/Startup.cs]: Baseadress comes from appsettings

File: SecretSanta/src/SecretSanta.Web/appsettings.Development.json [hunk 6]
```diff
@@ -5,5 +5,6 @@
       "Microsoft": "Warning",
       "Microsoft.Hosting.Lifetime": "Information"
     }
-  }
+  },
+  "ApiUrl": "https://localhost:44388/"
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/appsettings.Development.json]: Added base adress

File: SecretSanta/src/SecretSanta.Web/Api/Client.g.cs [hunk 7]
```diff
@@ -336,7 +371,7 @@ public async System.Threading.Tasks.Task DeleteAsync(int id, System.Threading.Ca
                 throw new System.ArgumentNullException("id");
 
             var urlBuilder_ = new System.Text.StringBuilder();
-            urlBuilder_.Append(BaseUrl != null ? BaseUrl.TrimEnd('/') : "").Append("/api/Gift/{id}");
+            urlBuilder_.Append("api/Gift/{id}");
             urlBuilder_.Replace("{id}", System.Uri.EscapeDataString(ConvertToString(id, System.Globalization.CultureInfo.InvariantCulture)));
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/Api/Client.g.cs]: Client.g is updated with proper settings

File: SecretSanta/src/SecretSanta.Web/Views/Shared/_Layout.cshtml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+<head>
+    <meta name="viewport" content="width=device-width" />
+    <title>@ViewBag.Title</title>
+</head>
+<body>
+    <div>
+        @RenderBody()
+    </div>
+</body>
+</html>
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/Views/Shared/_Layout.cshtml]: Your layout is here, but it doesn't have the controller links, although your template does. I believe running npm run build:dev on the command line in your web project should do the trick.

File: SecretSanta/src/SecretSanta.Web/Controllers/GiftsController.cs [hunk 9]
```diff
@@ -1,26 +1,80 @@
...
+            {
+                HttpClient httpClient = ClientFactory.CreateClient("SecretSantaApi");
+
+                var client = new GiftClient(httpClient);
+                var updatedGift = await client.PutAsync(id, giftInput);
+
+                result = RedirectToAction(nameof(Index));
+            }
+
+            return result;
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/Controllers/GiftsController.cs]: Did make create and edit. But your web page doesn't even display the current items on index. Look at the blogengine's index html file. Also didn't make delete.

Comment (StevenZuelke) [reply SecretSanta/src/SecretSanta.Web/Controllers/GiftsController.cs]: Your validation does work though, nice job

File: SecretSanta/src/SecretSanta.Web/Views/_ViewImports.cshtml [hunk 10]
```diff
@@ -1 +1,3 @@
-﻿@addTagHelper *, TagHelperPack
\ No newline at end of file
+﻿@using SecretSanta.Web.Api
+
+@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
```
Comment (StevenZuelke) [inline SecretSanta/src/SecretSanta.Web/Views/_ViewImports.cshtml]: Added taghelpers

File: SecretSanta/src/SecretSanta.Web/SecretSanta.Web.csproj [hunk 11]
```diff
@@ -1,11 +1,25 @@
...
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
     <PackageReference Include="TagHelperPack" Version="0.5.0-pre-20191217.4" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="wwwroot\" />
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (breaman) [inline SecretSanta/src/SecretSanta.Web/SecretSanta.Web.csproj]: we don't want this entry here since typescript is being compiled by webpack

**Pull request closed without merging**